dace/transformation/passes/resolve_condition.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2019-2024 ETH Zurich and the DaCe authors. All rights reserved.
+""" Resolve condition transformation """
+
+from typing import Any, Dict
+
+from dace.transformation import pass_pipeline as ppl
+import sympy as sy
+from dace.properties import CodeBlock
+from dace import sdfg as sd, symbolic
+from dace.properties import Property, make_properties, CodeBlock
+from sympy.parsing.sympy_parser import parse_expr
+from dace import symbolic
+from dace.frontend.python.astutils import unparse
+from dace.transformation.pass_pipeline import Pipeline
+import ast
+
+
+def eliminate_branch(sdfg: sd.SDFG, initial_edge: sd.graph.Edge):
```

Comment:
Looks like yes, it should become a reusable, documented utility function :-)

Reply:
Agreed! This one is a good candidate for "building block" used by multiple transformations